### PR TITLE
Update recipe development environment getting started guide

### PR DESCRIPTION
**Pull request by mike-solomon**
* Removes references to Kotlin that were no longer applicable
* Adds comments to confusing sections of code
* Adds missing Java dependency
* Cleans up project layout to be more accurate
* Provides a better explanation for local testing
* Minor grammar / spelling fixes

### CODE REVIEW

File: getting-started/recipe-development-environment.md [hunk 1]
```diff
@@ -111,6 +115,11 @@ dependencies {
         <artifactId>rewrite-java-11</artifactId>
         <scope>runtime</scope>
     </dependency>
+    <dependency>
+        <groupId>org.openrewrite</groupId>
+        <artifactId>rewrite-java-17</artifactId>
+        <scope>runtime</scope>
+    </dependency>
```
Comment (mike-solomon) [inline getting-started/recipe-development-environment.md:122]: I'm pretty sure we just forgot to add a `java 17` section here. Let me know if I misunderstood something, though.

File: getting-started/recipe-development-environment.md [hunk 2]
```diff
@@ -54,6 +54,8 @@ dependencies {
 
     // rewrite-java dependencies only necessary for Java Recipe development
     implementation("org.openrewrite:rewrite-java")
+
+    // add based on your Java needs
```
Comment (mike-solomon) [inline getting-started/recipe-development-environment.md:58]: I wanted to clarify that you don't need all of these for a given project.

Comment (sambsnyd) [reply getting-started/recipe-development-environment.md:58]: Technically all that's needed is the one corresponding to the JDK version you'll be running on. They can coexist on a classpath. So it's a bit redundant to have all three, but it's also maximally flexible for whatever JDK version they end up using.

Comment (sambsnyd) [reply getting-started/recipe-development-environment.md:58]: Technically all that's needed is the one corresponding to the JDK version you'll be running on. They can coexist on a classpath. So it's a bit redundant to have all three, but it's also maximally flexible for whatever JDK version they end up using. There's no need for users to spend effort considering this point if they accept the default. If we're going to have a comment explaining, which is fine, it should probably not imply that they need to consider and make a choice when no such consideration or choice is necessary.

**Review by sambsnyd — CHANGES_REQUESTED** [review]
Looks good overall

**Comment by mike-solomon** [comment]
Merging in the changes. Please feel free to let me know if I should change anything else or if the changes I made were incorrect :)

**Pull request merged**